FWCore/PluginManager/bin/refresh.cc

```diff
@@ -272,9 +272,13 @@ int main(int argc, char** argv) try {
         // Throw if any of the child died with non 0 status.
         int status = 0;
         waitpid(worker, &status, 0);
-        if (WIFEXITED(status) == true && status != 0) {
-          std::cerr << "Error while processing." << std::endl;
-          exit(status);
+        if (WIFEXITED(status) != 0 and WEXITSTATUS(status) != 0) {
+          std::cerr << "Error while processing: " << WEXITSTATUS(status) << std::endl;
+          exit(WEXITSTATUS(status));
+        }
+        if (WIFSIGNALED(status) != 0) {
+          std::cerr << "Got signal while processing: " << WTERMSIG(status) << std::endl;
+          exit(WTERMSIG(status));
```

Comment:
If you want to replicate the return value, you'd need to do `exit(128+WTERMSIG(status))` as `WTERMSIG` says it returns the signal number (e.g. for SegFault it would be 11) and not the exit code associated to that return value.

As we'd probably like to differentiate the main executable failing and a child failing, maybe we'd want a different offset from 128?

Reply:
A segfaulting child process actually gets caught by the `WIFEXITED(status) != 0` in above, i.e. it is apparently considered to be part of "normal termination" than "termination by signal". I'll add the `128+` there in any case (after we agree on the next point :) )

> As we'd probably like to differentiate the main executable failing and a child failing, maybe we'd want a different offset from 128?

Do we need to differentiate on that level in the exit code? I suppose scram itself only cares whether `edmPluginRefresh` succeeded or not. In the case of failure, we have the printouts above for the case where a child process fails (I could make the printouts more clear that it is the child process that fails). I.e. the build log should have information on whether the main process or a child process failed.

If we do, I'd use the same offset for the "normal termination" and "termination by signal" exit codes (i.e. `exit(WEXITSTATUS(status)+offset)` and `exit(128+WTERMSIG(status)+offset)`).

Reply:
On second thought, I think we don't need to distinguish between child or parent process return codes since we have error messages which do that work. Therefore I don't think we need `offset`.

Reply:
Last push adds the `128+` for the signal exit code, and text in the printouts to make it clear it was the child process that failed.